src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11258,6 +11258,7 @@ private ParameterSyntax ParseLambdaParameter(bool isFirst, ref bool hasTypes)
             var pk = this.PeekToken(1).Kind;
             if (isRefOrOutOrParams
                 || (pk != SyntaxKind.CommaToken && pk != SyntaxKind.CloseParenToken && (hasTypes || isFirst))
+                || (this.CurrentToken.Kind == SyntaxKind.OpenParenToken && pk == SyntaxKind.CloseParenToken && (hasTypes || isFirst))
```

Comment:
Factoring the conditions together made it rather hard to read. I added a comment as suggested.